src/vm/assembly.cpp

```diff
@@ -2588,6 +2588,7 @@ HRESULT RunMain(MethodDesc *pFD ,
             // Set the return value to 0 instead of returning random junk
             *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal = GetLatchedExitCode();
```

Comment:
Your built binaries may be out of synch.  I just changed this test in fc4497b4679c4282b9e30bedf2c20060033cf437 to use the supported Environment.Exit instead of Environment.ExitCode, which, though it seems to be present in mscorlib, is ignored.  Whether Environment.ExitCode should be implemented is perhaps an interesting question.  Apparently Environment.Exit wasn't initially supported by the .NET Core framework, but was added later, without adding the related Environment.ExitCode.